### DISCUSSION TITLE
Deactivated plugins

### DISCUSSION THREAD

**Discussion by waston4u** [apohl79/audiogridder#996] [Help]
I'm not sure if it is me or a bug, but when I deactivate some plugins, restart, or a new startup, it activates them again.

Any idea?

**Comment by apohl79** [comment]
Should not happen, so seems like a bug, but I can't reproduce this. They stay deactivated.

Can you check the server config file after you deactivated the plugins? There is a section, that should list the ones you deactivated:

```
    "ExcludePlugins": [
       <list of plugins>
    ],
```

If that is the case, and you start your server again, with the plugins being activated, can you provide the server logfile?

Check this link for the file locations: https://audiogridder.com/bug-report-diagnostics-locations/

**Comment by waston4u** [comment]
Hi Andreas

Attached you can find some logs and print screens.

Thank you
Miro

**Reply by apohl79** [reply]
No attachments. You probably used your email client, but you have to upload attachments via browser.

**Reply by waston4u** [reply]
Haha yeah, that's what exactly happened..;) I completely forgot it would strip it out.

[AFTER_AudioGridderServer_2022-08-23_09-19-11.log](https://github.com/apohl79/audiogridder/files/9413726/AFTER_AudioGridderServer_2022-08-23_09-19-11.log)
[BEFORE_AudioGridderServer_2022-08-23_09-15-42.log](https://github.com/apohl79/audiogridder/files/9413728/BEFORE_AudioGridderServer_2022-08-23_09-15-42.log)
![my_server_id_1_config](https://user-images.githubusercontent.com/17701976/186354901-16b37332-0d44-43eb-855c-9334d8f3e1b8.png)
![my_server_id_1_deactivated](https://user-images.githubusercontent.com/17701976/186354916-ab26da45-ca69-4167-a436-20b451cd98da.png)
![default_server_config](https://user-images.githubusercontent.com/17701976/186354934-8ba941ad-4171-4a24-a894-958707dbe21a.png)

**Reply by apohl79** [reply]
Ok I think something is messed up with the config files. Some versions back (beta10 I believe) I added a feature, that creates a dedicated config file for each server with an ID > 0. In your case `audiogridderserver1.cfg` as you have set the ID to 1. The problem seems to be, that when you upgraded, you were using ID 1 as the default ID (you have ID 1 in the default config `audiogridderserver.cfg`).

Unfortunately since that version, your default config will not be updated anymore, as the server reads the ID from the config file (and sets it to 1) and writes to a different file.

I will create a bug ticket for this.

There are two options to solve this:
1) Change the ID to 0 in the server settings and hit save. This will update the default config, but your server will have a new ID.
2) Start the server with the `-id 1` command line parameter (run: `open /Applications/AudioGridderServer.app --args -id 1`)
